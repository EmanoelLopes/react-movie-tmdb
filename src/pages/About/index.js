import React from 'react';

const About = () => (
  <React.Fragment>
    <h1>About</h1>
  </React.Fragment>
);

export default About;

