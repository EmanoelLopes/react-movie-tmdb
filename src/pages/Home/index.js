import React from 'react';
import { connect } from 'react-redux';
import { bool } from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilm, faTv } from '@fortawesome/free-solid-svg-icons';
import {
  MoviesContainer,
  TVSeriesContainer,
  HeroSectionContainer,
} from 'containers';
import Loading from 'components/Loading';
import { H1 } from 'utils/styled';

const Home = ({ loading }) => {
  return (
    <React.Fragment>
      {(loading) && <Loading />}
      <HeroSectionContainer />
      <div className="container">
        <H1>Popular Movies <FontAwesomeIcon icon={faFilm} /></H1>
        <MoviesContainer />
      </div>
      <div className="container">
        <H1>Popular TV Series <FontAwesomeIcon icon={faTv} /></H1>
        <TVSeriesContainer />
      </div>
    </React.Fragment>
  );
};

Home.propTypes = {
  loading: bool.isRequired,
};

const mapStateToProps = state => ({
  loading: state.loading,
});

export default connect(mapStateToProps)(Home);
