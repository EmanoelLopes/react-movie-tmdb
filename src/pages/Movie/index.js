import React from 'react';
import { H1 } from 'utils/styled';

const Movie = () => {
  return (
    <div className="movie">
      <div className="container">
        <H1>Movie</H1>
      </div>
    </div>
  );
};

export default Movie;
