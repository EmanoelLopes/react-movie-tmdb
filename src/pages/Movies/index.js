import React from 'react';

const Movies = () => {
  return (
    <div className="movies">
      <h1>Movies</h1>
    </div>
  );
};

export default Movies;
