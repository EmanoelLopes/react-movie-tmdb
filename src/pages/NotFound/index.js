import React from 'react';

const NotFound = () => (
  <React.Fragment>
    <h1>Not Found Page :/</h1>
  </React.Fragment>
);

export default NotFound;
