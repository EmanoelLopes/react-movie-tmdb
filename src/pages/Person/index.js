import React from 'react';

const Person = () => (
  <div className="person">
    <h1>Person</h1>
  </div>
);

export default Person;
