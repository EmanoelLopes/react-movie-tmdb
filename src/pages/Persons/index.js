import React from 'react';

const Persons = () => (
  <div className="persons">
    <h1>Persons</h1>
  </div>
);

export default Persons;
