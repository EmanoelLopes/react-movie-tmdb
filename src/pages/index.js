import { default as Home } from 'pages/Home';
import { default as About } from 'pages/About';
import { default as TVs } from 'pages/TVs';
import { default as TV } from 'pages/TV';
import { default as Movies } from 'pages/Movies';
import { default as Movie } from 'pages/Movies';
import { default as Persons } from 'pages/Persons';
import { default as Person } from 'pages/Person';
import { default as NotFound } from 'pages/NotFound';

export {
  Home,
  About,
  TVs,
  TV,
  Movies,
  Movie,
  Persons,
  Person,
  NotFound,
};
