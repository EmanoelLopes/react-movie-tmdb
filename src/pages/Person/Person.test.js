import React from 'react';
import { mount } from 'enzyme';
import { Person } from 'pages';
import Root from 'components/Root';

let wrapped;
let defaultProps = {};

beforeEach(() => {
  wrapped = mount(
    <Root>
      <Person {...defaultProps} />
    </Root>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Person />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
