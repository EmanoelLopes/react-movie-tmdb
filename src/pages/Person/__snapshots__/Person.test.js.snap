// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Person /> should match snapshot 1`] = `
<Root>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Person>
      <div
        className="person"
      >
        <h1>
          Person
        </h1>
      </div>
    </Person>
  </Provider>
</Root>
`;
